core/src/main/scala/io/finch/Bootstrap.scala

```diff
@@ -81,17 +118,27 @@ class Bootstrap[F[_], ES <: HList, CTS <: HList](
     ts.apply(endpoints, opts, ctx)
   }
 
-  def toService(implicit F: Async[F], ts: Compile[F, ES, CTS]): Resource[F, Service[Request, Response]] =
-    Endpoint.toService(compile)
+  def listen(address: String)(implicit F: Async[F], ts: Compile[F, ES, CTS]): Resource[F, ListeningServer] = {
+    val compiled = middleware(compile)
+
+    val serviced = Dispatcher[F].flatMap { dispatcher =>
+      Resource.make(F.pure(ToService(compiled, dispatcher))) { service =>
+        F.defer(service.close().toAsync[F])
+      }
+    }
+
+    serviced.flatMap { service =>
+      val filtered = filter.andThen(service)
+      Resource.make(F.pure(server.serve(address, filtered))) { listening =>
```

Comment:
It is an elegant solution, but it seems to me that there is no input for server configurations. (e.g: `withX` methods on `com.twitter.finagle.Http.Server`)

Reply:
I think what I envisioned is that a final version of the server is passed to Bootstrap.apply. A few of our examples leverage that already (e.g, the streaming one).

Reply:
Oh, I missed it! 👍